### PR TITLE
Introduce a Windbg debugger extension for UEFI platforms

### PR DESCRIPTION
**Pull request by cfernald**

### CODE REVIEW

File: UefiDbgExt/dbgexts.cpp [hunk 1]
```diff
@@ -0,0 +1,351 @@
+/*++
+
+Copyright (c) 2000  Microsoft Corporation
```
Comment (makubacki) [inline UefiDbgExt/dbgexts.cpp:3]: I assume some of these filers are old boilerplate but the copyright headers are mixed and with various flavors of Microsoft copyright and some are missing `SPDX-License-Identifier: BSD-2-Clause-Patent`.

I suggest we clean them up and make them consistent within this code tree.

**Comment by makubacki** [comment]
Can you please add a Readme to the UefiDbgExt directory that describes information like applicability, limitations, basic usage, and theory of operation (e.g. memory scanning details, etc.)?

**Comment by cfernald** [comment]
> Can you please add a Readme to the UefiDbgExt directory that describes information like applicability, limitations, basic usage, and theory of operation (e.g. memory scanning details, etc.)?

Hey, there is already a readme, but I'll add some of the details you mentioned!

File: UefiDbgExt/memory.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Module Name:
+
+    memory.cpp
+
+Abstract:
+
+    This file contains debug commands for memory operations.
+
+    SPDX-License-Identifier: BSD-2-Clause-Patent
```
Comment (cfernald) [inline UefiDbgExt/memory.cpp:15]: double SPDX

File: UefiDbgExt/readme.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+The project can also be loaded and built in Visual Studio using the solution
+file. This project requires the Windows SDK and the Windows Driver Kit.
+
+## Installing the Extension
+
+Debugger extensions can be loaded into windbg several ways. First, by
+manually loading once already in windbg. This can be done with the .load
+command. Though this will not persist across windbg sessions.
+
+    .load <path to uefiext.dll>
```
Comment (kuqin12) [inline UefiDbgExt/readme.md:27]: Do we publish this as an artifact? If so, should we add a url to point to where they are?

**Review by kuqin12 — APPROVED** [review]

File: UefiDbgExt/readme.md (hunk 3 above)
Comment (cfernald) [reply UefiDbgExt/readme.md:27]: Its uploaded as an artifact of the github action. which unfortunately cannot be linked in a direct way for the most recent version that I have found yes, but i'll add this information tot he readme.  e.g. https://github.com/cfernald/mu_tiano_platforms/actions/runs/3371109346

Comment (cfernald) [reply UefiDbgExt/readme.md:27]: after thinking a bit more, we will probably just publish it along with the firmware (probably to nuget) when we setup the release pipeline.

**Comment by makubacki** [comment]
> > Can you please add a Readme to the UefiDbgExt directory that describes information like applicability, limitations, basic usage, and theory of operation (e.g. memory scanning details, etc.)?
> 
> Hey, there is already a readme, but I'll add some of the details you mentioned!

Thanks [UefiDbgExt/readme.md](https://github.com/microsoft/mu_tiano_platforms/pull/118/files#diff-6df09dedaac96dec966b22cca0fd388808a823c55204c228252ada78d8e7f350) covers what I was looking for

**Review by makubacki — APPROVED** [review]

**Comment by out0xb2** [comment]
This is super-cool!  Still a fan from afar!

**Pull request merged**